### PR TITLE
feat: display search filters in map layer config panel

### PR DESCRIPTION
**Pull request by ruanyl**
Adding filters to map layer which allows user to do more accurate and customized searches

Signed-off-by: Yulong Ruan <ruanyl@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by junqiu-lei** [comment]
Great thanks @ruanyl, I just found one warning message for this PR code on my local console when I input boolean value on my local. 

reproduce:

image



```
Warning: Failed prop type: Invalid prop `value` supplied to `EuiSelect`, expected one of type [string, number].
    in EuiSelect (created by ValueInputTypeUI)
    in ValueInputTypeUI (created by InjectIntl(ValueInputTypeUI))
    in InjectIntl(ValueInputTypeUI) (created by PhraseValueInputUI)
    ......
```

File: maps_dashboards/public/components/layer_control_panel/layer_control_panel.tsx [hunk 1]
```diff
@@ -83,12 +85,17 @@ const LayerControlPanel = memo(({ maplibreRef, setLayers, layers }: Props) => {
         if (sourceConfig.showTooltips === true && sourceConfig.tooltipFields.length > 0) {
           sourceFields.push(...sourceConfig.tooltipFields);
         }
+        let indexPattern: IndexPattern | undefined = undefined;
```
Comment (junqiu-lei) [inline maps_dashboards/public/components/layer_control_panel/layer_control_panel.tsx:88]:
```suggestion
        let indexPattern: IndexPattern | undefined;
```

**Comment by ruanyl** [comment]
@junqiu-lei nice catch on the EuiInput warning! But it looks like it's Eui related, it actually happen to EuiSelect component on other places as well, like on `Discover` page, when you add the same filter, you could see the same warning(to see the warning, try to refresh the page and start from scratch).

**Review by junqiu-lei — APPROVED** [review]
LGTM, thanks!

**Review by VijayanB — APPROVED** [review]

**Pull request merged**